test/extensions/filters/http/adaptive_concurrency/adaptive_concurrency_filter_test.cc

```diff
@@ -252,6 +253,25 @@ TEST_F(AdaptiveConcurrencyFilterTest, EncodeHeadersValidTest) {
   filter_->encodeComplete();
 }
 
+TEST_F(AdaptiveConcurrencyFilterTest, DisregardHealthChecks) {
+  StreamInfo::MockStreamInfo stream_info;
+  EXPECT_CALL(decoder_callbacks_, streamInfo()).WillOnce(ReturnRef(stream_info));
+  EXPECT_CALL(stream_info, healthCheck()).WillOnce(Return(true));
+
+  Http::TestHeaderMapImpl request_headers;
+
+  // We do not expect a call to forwardingDecision() during decode.
+
+  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_->decodeHeaders(request_headers, true));
+
+  Http::TestHeaderMapImpl response_headers;
+  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_->encodeHeaders(response_headers, false));
+
+  // We do not expect a call to recordLatencySample() as well.
+
+  filter_->encodeComplete();
```

Comment:
nit: encodeComplete would not be called unless there is a header only response (end_stream = true). Can you try to make this more realistic here and elsewhere in the tests? This doesn't matter right now but it may matter later depending on filter changes.

Reply:
Swapped it so all `encodeHeaders` calls have end_stream=true. Let me know if that's what you had in mind.